lib/src/resource/base.dart

```diff
@@ -26,12 +26,22 @@ class Subtype {
   const Subtype(this.namespace, this.resourceType, this.resourceSubtype);
 
   /// Create a new [Subtype] from a [ResourceName]
+  ///
+  /// ```
+  /// // Example, creating a Sensor from a resource named "mySensor":
+  /// var mySensor = Sensor.fromResourceName('mySensor');
+  /// ```
```

Comment:
```suggestion
```
Should these actually not be here since they're not part of the API?

Reply:
There may be considerations I'm not thinking of but it seems harmless and more thorough to include these.